backend/src/main/java/com/google/sps/datastore/UserDatastore.java

```diff
@@ -27,11 +46,15 @@ public UserDatastore(DatastoreService datastore) {
   /** Put user in datastore. Overwrite user entity if user with same username already exists. */
   public void addUser(User user) {
     // Set properties of entity based on user fields
-    Entity userEntity = new Entity(KIND_USER, user.getUsername());
-    userEntity.setProperty(PROPERTY_USERNAME, user.getUsername());
+    Entity entity = new Entity(KIND_USER, user.getUsername());
+    entity.setProperty(PROPERTY_USERNAME, user.getUsername());
+    entity.setProperty(PROPERTY_DURATION, user.getDuration());
+    entity.setProperty(PROPERTY_ROLE, user.getRole());
+    entity.setProperty(PROPERTY_PRODUCT_AREA, user.getProductArea());
+    entity.setProperty(PROPERTY_MATCH_PREFERENCE, user.getMatchPreference().getValue());
```

Comment:
WDYT about extracting a method to convert from a User to an entity, parallel to getUserFromEntity below?